code/game/machinery/portable_turret.dm

```diff
@@ -538,6 +546,24 @@ var/global/list/turret_icons
 
 	return H.assess_perp(src, check_access, check_weapons, check_records, check_arrest)
 
+/obj/machinery/porta_turret/proc/assess_mechs(obj/mecha/M)
+	if(!M.occupant)
+		return TURRET_NOT_TARGET //dont shoot empty guys, maybe the HOS parked it there "just in case"
+
+	if(!check_n_synth)
+		return TURRET_NOT_TARGET // targeting disabled
```

Comment:
```suggestion
	if(!check_n_synth)
		return TURRET_NOT_TARGET // targeting disabled
	
	if(!M.occupant)
		return TURRET_NOT_TARGET //dont shoot empty guys, maybe the HOS parked it there "just in case"
```
в целом подобные проверки надо делать от самого высоковероятного и дешевого к проверке к самому маловероятному и дорогому к проверке, тут имхо чаще ИИ будет не врубать режим уничтожения мехов(хотя если он по дефолту, то лучше оставить так как было), а потом уже проверять на водителя меха

Reply:
Режим удаления мехов это режим который говорит убивать все что не борги и ИИ, то есть он по дефолту врублен на всех турелях

Reply:
а так как туррели для этого и задуманы, вырубать этот режим было бы странно

Reply:
тогда ок резолвай

Reply:
блин на гитхабе можно резолвать если не изменены строчки были чет не пойму как в гитлабе просто кнопка резолв и все